### PR TITLE
Refactor `KafkaAssemblyMockOperatorTest` to use StrimziPodset 

### PR DESCRIPTION
**Pull request by ShubhamRwt**
Signed-off-by: ShubhamRwt <srawat@redhat.com>

### Type of change

_Select the type of your PR_
- Refactoring

### Description

This PR address the issue mentioned in #6903. Several Parameters which are not required are removed form the class like resource and storage. This PR also tend to remove the Stateful set used in the test. The parametized tests are changed to normal test and now all tests run on a single kafka configuration

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java [hunk 1]
```diff
@@ -546,16 +428,18 @@ public void testReconcileUpdatesKafkaStorageType(Params params, VertxTestContext
...
                         .orElse(new ArrayList<>()));
-                originalVolumes.set(Optional.ofNullable(client.apps().statefulSets().inNamespace(NAMESPACE).withName(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME)).get())
-                        .map(StatefulSet::getSpec)
-                        .map(StatefulSetSpec::getTemplate)
-                        .map(PodTemplateSpec::getSpec)
-                        .map(PodSpec::getVolumes)
-                        .orElse(new ArrayList<>()));
-                originalInitContainers.set(Optional.ofNullable(client.apps().statefulSets().inNamespace(NAMESPACE).withName(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME)).get())
+
+                originalVolumes.set(Optional.ofNullable(supplier.strimziPodSetOperator.client().inNamespace(NAMESPACE).withName(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME)).get())
```
Comment (ShubhamRwt) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java:438]: @scholzj This is the part where Iam majorly stuck at.

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java [hunk 2]
```diff
@@ -230,46 +124,44 @@ public static void after() {
      * since this is a parameterized set, the tests params are only available at test start
      * This must be called before each test
      */
-    private void init(Params params) {
-        setFields(params);
+    private void init() {
```
Comment (scholzj) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java:127]: Since we removed the parametrization, should this change to `@BeforeEach`?

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java [hunk 3]
```diff
@@ -283,10 +175,14 @@ private void init(Params params) {
         mockKube.start();
 
         PlatformFeaturesAvailability pfa = new PlatformFeaturesAvailability(false, kubernetesVersion);
-        ResourceOperatorSupplier supplier = supplierWithMocks();
-        ClusterOperatorConfig config = ResourceUtils.dummyClusterOperatorConfig(VERSIONS, ClusterOperatorConfig.DEFAULT_OPERATION_TIMEOUT_MS, "-UseStrimziPodSets");
+        supplier = supplierWithMocks();
+        podSetController = new StrimziPodSetController(NAMESPACE, Labels.EMPTY, supplier.kafkaOperator, supplier.strimziPodSetOperator, supplier.podOperations, ClusterOperatorConfig.DEFAULT_POD_SET_CONTROLLER_WORK_QUEUE_SIZE);
+        podSetController.start();
```
Comment (scholzj) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java:180]: You should stop it at the end of each test probably.

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java (hunk 1 above)
Comment (scholzj) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java:438]: The VolumeClaimTemplates are gone now. So there are only two things to check:
* The actual PVCs created by Strimzi
* The volumes of the pods in the StrimziPodSet (each pod should mount the right PVC)

I think that the `assertVolumes` method should be updated to cover the second part. The `PodSetUtils` class has utility methods for decoding the Pods from the StrimziPodSet resource to make it easy to check things. 

The `assertPVC` method should cover the first. It needs to be rewritten to check the PVCs and not the PersistentVolumeClaimTemplates.

TBH, I do not entirely understand how do the init containers fit here. IMHO they have nothing to do with persistent or ephemeral storage. But that should be easy to fix again by decoding the Pods from the StrimziPodSet and checking the pods.

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java [hunk 4]
```diff
@@ -230,45 +121,45 @@ public static void after() {
...
+                         .withStorageClass("foo")
+                         .withDeleteClaim(true)
+                         .endPersistentClaimStorage()
+                         .endZookeeper()
+                 .endSpec()
+                 .build();
+
+        // Configure replicas and storage
+        this.kafkaReplicas = cluster.getSpec().getKafka().getReplicas();
+        this.kafkaStorage = cluster.getSpec().getKafka().getStorage();
```
Comment (scholzj) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java:162]: You know what the number of replicas and storage is so you can probably define it is `static final` constants?

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java [hunk 5]
```diff
@@ -296,42 +190,29 @@ private ResourceOperatorSupplier supplierWithMocks() {
...
-    }
 
     private Future<Void> initialReconcile(VertxTestContext context) {
         LOGGER.info("Reconciling initially -> create");
         return operator.reconcile(new Reconciliation("test-trigger", Kafka.RESOURCE_KIND, NAMESPACE, CLUSTER_NAME))
             .onComplete(context.succeeding(v -> context.verify(() -> {
-                StatefulSet kafkaSts = client.apps().statefulSets().inNamespace(NAMESPACE).withName(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME)).get();
-                kafkaSts.setStatus(new StatefulSetStatus());
+                StrimziPodSet sps = supplier.strimziPodSetOperator.client().inNamespace(NAMESPACE).withName(CLUSTER_NAME + "-kafka").get();
+                sps.setStatus(new StrimziPodSetStatus());
```
Comment (scholzj) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java:199]: Why do we need to set this?

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java [hunk 6]
```diff
@@ -296,42 +190,29 @@ private ResourceOperatorSupplier supplierWithMocks() {
...
+                StrimziPodSet zkSps = supplier.strimziPodSetOperator.client().inNamespace(NAMESPACE).withName(CLUSTER_NAME + "-zookeeper").get();
+                zkSps.getSpec().getPods().stream().map(PodSetUtils::mapToPod).forEach(pod -> {
+                    assertThat(pod.getMetadata().getAnnotations(), hasEntry(Ca.ANNO_STRIMZI_IO_CLUSTER_CA_CERT_GENERATION, "0"));
+                });
 
-                assertThat(client.configMaps().inNamespace(NAMESPACE).withName(KafkaResources.kafkaMetricsAndLogConfigMapName(CLUSTER_NAME)).get(), is(notNullValue()));
+                //Removed in StrimziPodSets
+                //assertThat(client.configMaps().inNamespace(NAMESPACE).withName(KafkaResources.kafkaMetricsAndLogConfigMapName(CLUSTER_NAME)).get(), is(notNullValue()));
                 assertThat(client.configMaps().inNamespace(NAMESPACE).withName(KafkaResources.zookeeperMetricsAndLogConfigMapName(CLUSTER_NAME)).get(), is(notNullValue()));
-                assertResourceRequirements(context, KafkaResources.kafkaStatefulSetName(CLUSTER_NAME));
```
Comment (scholzj) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java:334]: Why don't we need to assert the resources anymore?

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java [hunk 7]
```diff
@@ -296,42 +190,29 @@ private ResourceOperatorSupplier supplierWithMocks() {
...
-                assertThat(zkSts.getSpec().getTemplate().getMetadata().getAnnotations(), hasEntry(StatefulSetOperator.ANNO_STRIMZI_IO_GENERATION, "0"));
-                assertThat(zkSts.getSpec().getTemplate().getMetadata().getAnnotations(), hasEntry(Ca.ANNO_STRIMZI_IO_CLUSTER_CA_CERT_GENERATION, "0"));
+                StrimziPodSet zkSps = supplier.strimziPodSetOperator.client().inNamespace(NAMESPACE).withName(CLUSTER_NAME + "-zookeeper").get();
+                zkSps.getSpec().getPods().stream().map(PodSetUtils::mapToPod).forEach(pod -> {
+                    assertThat(pod.getMetadata().getAnnotations(), hasEntry(Ca.ANNO_STRIMZI_IO_CLUSTER_CA_CERT_GENERATION, "0"));
+                });
 
-                assertThat(client.configMaps().inNamespace(NAMESPACE).withName(KafkaResources.kafkaMetricsAndLogConfigMapName(CLUSTER_NAME)).get(), is(notNullValue()));
+                //Removed in StrimziPodSets
+                //assertThat(client.configMaps().inNamespace(NAMESPACE).withName(KafkaResources.kafkaMetricsAndLogConfigMapName(CLUSTER_NAME)).get(), is(notNullValue()));
```
Comment (scholzj) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java:214]: Why is this not needed anymore?

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java [hunk 8]
```diff
@@ -296,42 +190,29 @@ private ResourceOperatorSupplier supplierWithMocks() {
...
+                });
 
-                StatefulSet zkSts = client.apps().statefulSets().inNamespace(NAMESPACE).withName(KafkaResources.zookeeperStatefulSetName(CLUSTER_NAME)).get();
-                assertThat(zkSts, is(notNullValue()));
-                assertThat(zkSts.getSpec().getTemplate().getMetadata().getAnnotations(), hasEntry(StatefulSetOperator.ANNO_STRIMZI_IO_GENERATION, "0"));
-                assertThat(zkSts.getSpec().getTemplate().getMetadata().getAnnotations(), hasEntry(Ca.ANNO_STRIMZI_IO_CLUSTER_CA_CERT_GENERATION, "0"));
+                StrimziPodSet zkSps = supplier.strimziPodSetOperator.client().inNamespace(NAMESPACE).withName(CLUSTER_NAME + "-zookeeper").get();
+                zkSps.getSpec().getPods().stream().map(PodSetUtils::mapToPod).forEach(pod -> {
+                    assertThat(pod.getMetadata().getAnnotations(), hasEntry(Ca.ANNO_STRIMZI_IO_CLUSTER_CA_CERT_GENERATION, "0"));
+                });
```
Comment (scholzj) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java:211]: Should we test these for Zoo as well?

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java [hunk 9]
```diff
@@ -522,52 +383,21 @@ private Resource<Kafka> kafkaAssembly(String namespace, String name) {
...
-                        .orElse(new ArrayList<>()));
-
                 // Update the storage type
                 // ephemeral -> persistent
                 // or
                 // persistent -> ephemeral
                 Kafka updatedStorageKafka = null;
-                if (kafkaStorage instanceof EphemeralStorage) {
+                System.out.println(cluster.getSpec().getKafka().getStorage().getClass());
+                if (cluster.getSpec().getKafka().getStorage() instanceof EphemeralStorage) {
```
Comment (scholzj) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java:400]: You seem to always start with persistent storage. So why is this needed? It seems like it doesn't do anything.

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java [hunk 10]
```diff
@@ -589,68 +419,64 @@ public void testReconcileUpdatesKafkaStorageType(Params params, VertxTestContext
...
+            List<PersistentVolumeClaimVolumeSource> persistentVolumeClaim = new ArrayList<>();
+            // We need to loop through the pods to make sure they have the right values
+            sps.getSpec().getPods().stream().map(PodSetUtils::mapToPod).forEach(pod -> {
+                pod.getSpec().getVolumes().forEach(volume -> {
+                    if (volume.getPersistentVolumeClaim() != null) {
+                        persistentVolumeClaim.add(volume.getPersistentVolumeClaim());
+                    }
+                });
+            });
+            assertThat(persistentVolumeClaim.size(), is(3));
```
Comment (scholzj) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java:449]: These are not PVCs. These are volumes in the Pods / containers. You need to assert the actual PersistentVolumeClaim Kubernetes resources.

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java [hunk 11]
```diff
@@ -734,7 +557,7 @@ public void testReconcileKafkaScaleDown(Params params, VertxTestContext context)
             })))
             .compose(v -> operator.reconcile(new Reconciliation("test-trigger", Kafka.RESOURCE_KIND, NAMESPACE, CLUSTER_NAME)))
             .onComplete(context.succeeding(v -> context.verify(() -> {
-                assertThat(client.apps().statefulSets().inNamespace(NAMESPACE).withName(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME)).get().getSpec().getReplicas(),
+                assertThat(supplier.strimziPodSetOperator.client().inNamespace(NAMESPACE).withName(CLUSTER_NAME + "-kafka").get().getSpec().getPods().size(),
```
Comment (scholzj) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java:560]: You can use `Crds.strimziPodsetOperation(client)` instead of this. Much nicer and works everywhere even without the supplier.

**Review by scholzj — COMMENTED** [review]
Great that the tests are passing now. I left some comments.

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java (hunk 6 above)
Comment (ShubhamRwt) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java:334]: I removed the resource parameter and its related test since we discussed on chat that resource parameter might not be imp here since it has not much to do with MockKube.

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java (hunk 7 above)
Comment (ShubhamRwt) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java:214]: I checked the confgimaps generated after the deployment and I was not able to find the configmap related to this

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java (hunk 5 above)
Comment (ShubhamRwt) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java:199]: Resolved

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java (hunk 4 above)
Comment (ShubhamRwt) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java:162]: Resolved: I kept the storage parameter though since its being used throughout the code

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java [hunk 12]
```diff
@@ -230,45 +121,45 @@ public static void after() {
...
+                     .withNewZookeeper()
+                         .withReplicas(3)
+                        .withNewPersistentClaimStorage()
+                         .withSize("123")
+                         .withStorageClass("foo")
+                         .withDeleteClaim(true)
+                         .endPersistentClaimStorage()
+                         .endZookeeper()
+                 .endSpec()
+                 .build();
```
[the remark this answers is not in the document]
Comment (ShubhamRwt) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java:158]: Resolved

**Comment by scholzj** [comment]
@ShubhamRwt Your build seems to be failing with some checkstyle errors.

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java [hunk 13]
```diff
@@ -587,70 +392,59 @@ public void testReconcileUpdatesKafkaStorageType(Params params, VertxTestContext
...
         context.verify(() -> {
-            StatefulSet statefulSet = client.apps().statefulSets().inNamespace(NAMESPACE).withName(statefulSetName).get();
-            assertThat(statefulSet, is(notNullValue()));
-            List<Volume> volumes = statefulSet.getSpec().getTemplate().getSpec().getVolumes();
-            assertThat(originalVolumes.size(), is(volumes.size()));
-            assertThat(originalVolumes, is(volumes));
+            StrimziPodSet sps = supplier.strimziPodSetOperator.client().inNamespace(NAMESPACE).withName(podSetName).get();
+            assertThat(sps, is(notNullValue()));
+            List<PersistentVolumeClaim> pvc = client.persistentVolumeClaims().inNamespace(NAMESPACE).list().getItems();
+            assertThat(pvc.size(), is(originalSize));
```
Comment (scholzj) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java:417]: This is not exactly nice since you:

Get the PodSet and check its not null ... Why? Does not seem to belong here.
Check all PVCs for single size ... you should check only the PVCs which belong to this PodSet

Comment (scholzj) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java:417]: You should check more then just the count. You should check that its parsistent storage in the volume and not ephemeral storage.

Comment (ShubhamRwt) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java]: I have pushed the changes for the suggestion but I am little confused on getting PVC corresponding to the PodSet.

Comment (scholzj) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java]: You can for example filter it by name => they will start with the name of the pod / podset.

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java [hunk 14]
```diff
@@ -587,70 +389,62 @@ public void testReconcileUpdatesKafkaStorageType(Params params, VertxTestContext
...
-    private void assertVolumes(VertxTestContext context, String statefulSetName, List<Volume> originalVolumes) {
+    private void assertPVCs(VertxTestContext context, String podSetName) {
         context.verify(() -> {
-            StatefulSet statefulSet = client.apps().statefulSets().inNamespace(NAMESPACE).withName(statefulSetName).get();
-            assertThat(statefulSet, is(notNullValue()));
-            List<Volume> volumes = statefulSet.getSpec().getTemplate().getSpec().getVolumes();
-            assertThat(originalVolumes.size(), is(volumes.size()));
-            assertThat(originalVolumes, is(volumes));
+            assertThat(kafkaStorage.getType(), is("persistent-claim"));
+            List<PersistentVolumeClaim> pvc = new ArrayList<>();
```
Comment (ShubhamRwt) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java:411]: Is this correct now @scholzj ?

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java [hunk 15]
```diff
@@ -587,70 +389,62 @@ public void testReconcileUpdatesKafkaStorageType(Params params, VertxTestContext
...
+            client.persistentVolumeClaims().inNamespace(NAMESPACE).list().getItems().stream().forEach(persistentVolumeClaim -> {
+                if (persistentVolumeClaim.getMetadata().getName().contains(podSetName)){
```
Comment (scholzj) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java:413]: `startsWith`?

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java [hunk 16]
```diff
@@ -587,70 +389,62 @@ public void testReconcileUpdatesKafkaStorageType(Params params, VertxTestContext
...
+                    pvc.add(persistentVolumeClaim);
+                }
+            });
+            assertThat(pvc.size(), is(3));
```
Comment (scholzj) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java:417]: Aren't there more properties to check? Storage class etc.?

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java (hunk 14 above)
Comment (scholzj) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java:411]: Some more details, but yes, it does look better.

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java (hunk 16 above)
Comment (ShubhamRwt) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java:424]: I have added the assertion for storage class

Comment (scholzj) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java:424]: Maybe you can check also for example size? The `println` can be maybe removed and I think you have some checkstyle errors.

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java (hunk 15 above)
Comment (ShubhamRwt) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java]: The name starts with data followed by the `podSetName`. I have changed the test accordingly

**Review by scholzj — APPROVED** [review]
Thanks.

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java [hunk 17]
```diff
@@ -296,42 +189,24 @@ private ResourceOperatorSupplier supplierWithMocks() {
...
-                assertThat(kafkaSts.getSpec().getTemplate().getMetadata().getAnnotations(), hasEntry(Ca.ANNO_STRIMZI_IO_CLUSTER_CA_CERT_GENERATION, "0"));
-                assertThat(kafkaSts.getSpec().getTemplate().getMetadata().getAnnotations(), hasEntry(Ca.ANNO_STRIMZI_IO_CLIENTS_CA_CERT_GENERATION, "0"));
-
-                StatefulSet zkSts = client.apps().statefulSets().inNamespace(NAMESPACE).withName(KafkaResources.zookeeperStatefulSetName(CLUSTER_NAME)).get();
-                assertThat(zkSts, is(notNullValue()));
-                assertThat(zkSts.getSpec().getTemplate().getMetadata().getAnnotations(), hasEntry(StatefulSetOperator.ANNO_STRIMZI_IO_GENERATION, "0"));
-                assertThat(zkSts.getSpec().getTemplate().getMetadata().getAnnotations(), hasEntry(Ca.ANNO_STRIMZI_IO_CLUSTER_CA_CERT_GENERATION, "0"));
-
-                assertThat(client.configMaps().inNamespace(NAMESPACE).withName(KafkaResources.kafkaMetricsAndLogConfigMapName(CLUSTER_NAME)).get(), is(notNullValue()));
+                StrimziPodSet sps = supplier.strimziPodSetOperator.client().inNamespace(NAMESPACE).withName(CLUSTER_NAME + "-kafka").get();
```
[the remark this answers is not in the document]
Comment (scholzj) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java:197]: It uses the method for the StatefulSet name since they are the same. I personally would not add more methods at this point. But using the statefulset one here is a good point.

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java [hunk 18]
```diff
@@ -296,42 +189,24 @@ private ResourceOperatorSupplier supplierWithMocks() {
...
-            StatefulSet statefulSet = client.apps().statefulSets().inNamespace(NAMESPACE).withName(statefulSetName).get();
-            assertThat(statefulSet, is(notNullValue()));
-            ResourceRequirements requirements = statefulSet.getSpec().getTemplate().getSpec().getContainers().get(0).getResources();
-            if (resources != null && resources.getRequests() != null) {
-                assertThat(requirements.getRequests(), hasEntry("cpu", resources.getRequests().get("cpu")));
-                assertThat(requirements.getRequests(), hasEntry("memory", resources.getRequests().get("memory")));
-            }
-            if (resources != null && resources.getLimits() != null) {
-                assertThat(requirements.getLimits(), hasEntry("cpu", resources.getLimits().get("cpu")));
-                assertThat(requirements.getLimits(), hasEntry("memory", resources.getLimits().get("memory")));
```
[the remark this answers is not in the document]
Comment (ShubhamRwt) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java:310]: I removed it since when I was discussing about the changes with Jakub on the caht, we discussed that the resources play no role with the Mockkube so they were not of much use

**Review by tombentley — APPROVED** [review]
